core/dbt/common/helper_types.py

```diff
@@ -67,9 +68,13 @@ def _validate_items(self, items: List[str]):
 
 class WarnErrorOptions(IncludeExclude):
     def _validate_items(self, items: List[str]):
-        valid_exception_names = set(
+        valid_common_exception_names = set(
             [name for name, cls in dbt_event_types.__dict__.items() if isinstance(cls, type)]
         )
+        valid_core_exception_names = set(
+            [name for name, cls in core_dbt_event_types.__dict__.items() if isinstance(cls, type)]
+        )
+        valid_exception_names = valid_common_exception_names.union(valid_core_exception_names)
```

Comment:
thinking about this more.. I think we should also explicitly include adapter warnings here + add a test for it. 

we document the ability to use warn error options with an adapter event here: https://docs.getdbt.com/reference/global-configs/warnings